tests/frame/select_test.py

```diff
@@ -29,12 +32,22 @@ def test_non_string_select() -> None:
     pd.testing.assert_frame_equal(result, expected)
 
 
-def test_non_string_select_invalid() -> None:
+def test_int_select_pandas() -> None:
     df = nw.from_native(pd.DataFrame({0: [1, 2], "b": [3, 4]}))
-    with pytest.raises(TypeError, match="\n\nHint: if you were trying to select"):
+    with pytest.raises(InvalidIntoExprError):
         nw.to_native(df.select(0))  # type: ignore[arg-type]
 
 
+@pytest.mark.parametrize("invalid_select", [None, 0])
+def test_invalid_select(
+    constructor: Constructor, invalid_select: Any, request: pytest.FixtureRequest
+) -> None:
+    if "polars" in str(constructor):
+        request.applymarker(pytest.mark.xfail)
```

Comment:
anything that is not a `str` or `IntoExpr` become a literal in polars.
We are not blocking this behaviour at the moment. Is it something you think we should address for consistency sake?